### PR TITLE
CADC-12765 update cavern README for new config format

### PR DESCRIPTION
**Pull request by jburke-cadc**

### CODE REVIEW

File: cavern/README.md [hunk 1]
```diff
@@ -1,44 +1,100 @@
...
+### cavern.properties
+A `cavern.properties` file in /config is required to run this service.  The following keys are required:
 ```
-docker run -d --volume=/path/to/external/config:/config:ro --name cavern-tomcat images.canfar.net/skaha-system/cavern:latest
+# service identity
+org.opencadc.cavern.resourceID = ivo://{authority}/{name}
+
+# base directory for cavern files
+org.opencadc.cavern.filesystem.root = {directory path}
+
```
Comment (pdowler) [inline cavern/README.md:54]: let's change the above to
```
org.opencadc.cavern.filesystem.baseDir = {persistent data directory in container}
org.opencadc.cavern.filesystem.subPath = {relative path to the node/file content that could be mounted in other containers}
```

then remove the probe config and hard-code the subdir; for now we can hard code the `tomcat` for those tests

**Review by pdowler — CHANGES_REQUESTED** [review]

**Comment by pdowler** [comment]
Another thing that is missing from the config:
```
org.opencadc.cavern.privateKey = 
org.opencadc.cavern.publicKey = 
```

these are currently hard-coded in one or more places but should be configurable and *optional*. If not present, cavern cannot generate pre-auth URLs, but it can generate plain URLs.

File: cavern/README.md [hunk 2]
```diff
@@ -1,44 +1,100 @@
...
+Example:
 ```
-# database connection pools
-org.opencadc.cavern.uws.maxActive={max connections for jobs pool}
-org.opencadc.cavern.uws.username={database username for jobs pool}
-org.opencadc.cavern.uws.password={database password for jobs pool}
-org.opencadc.cavern.uws.url=jdbc:postgresql://{server}/{database}
+users = {user identity}
 ```
+`users` specifies the user(s) who are authorized to make calls to the service. The value is a list of user identities
```
[the remark this answers is not in the document]
Comment (jburke-cadc) [reply cavern/README.md:85]: The code change is minimal, a couple of lines.
Then update the README, which is copy the text from cavern into cadc-vosi.
Not a lot of work.
Ignoring all the other services that would need to be updated.

[the remark this answers is not in the document]
Comment (jburke-cadc) [reply cavern/README.md:85]: Removed the availability state text, and added an availabilityProperties init-param to the AvailabilityServlet in the web.xml with cadc-vosi.properties as the value.

File: cavern/README.md [hunk 3]
```diff
@@ -1,44 +1,100 @@
...
+(X500 distinguished name), one line per user. Optional: if the `cavern-availability.properties` is not found or does not
+list any `users`, the service will function in the default mode (ReadWrite) and the state will not be changeable.
 
-The `uws` pool manages (create, alter, drop) uws tables and manages the uws content (creates and modifies jobs in the uws
-schema when jobs are created and executed by users.
\ No newline at end of file
+## building
+```
```
[the remark this answers is not in the document]
Comment (jburke-cadc) [reply cavern/README.md:90]: Updated

File: cavern/README.md (hunk 1 above)
Comment (jburke-cadc) [reply cavern/README.md:54]: Updated the README, will update the tests.

File: cavern/README.md [hunk 4]
```diff
@@ -1,37 +1,35 @@
-# Cavern - A mountable, filesystem-based VOSpace implementation
+# Mountable, filesystem-based VOSpace-2.1 service (cavern)
+`cavern` is a VOSpace 2.1 compliant service where both the data and metadata are based on an underlying file system.  
+The permissions, too, are governed by the file system permissions.  For the users and groups in the file system 
+to correspond to the users in the VOSpace REST API, a SSSD connection is made from the file system to LDAP.
```
Comment (pdowler) [inline cavern/README.md:4]: still says SSSD here

File: cavern/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (pdowler) [reply cavern/README.md]: good for now; cadc-vosi changes out of scope as discussed

**Review by pdowler — APPROVED** [review]

**Pull request merged**